app/jobs/regular/open_topic.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Jobs
+  class OpenTopic < ::Jobs::Base
+    def execute(args)
+      topic_timer = TopicTimer.find_by(id: args[:topic_timer_id])
+      return if !topic_timer&.runnable?
+
+      topic = topic_timer.topic
+      user = topic_timer.user
+
+      if topic.blank?
+        topic_timer.destroy!
+        return
+      end
+
+      if !Guardian.new(user).can_open_topic?(topic)
+        topic_timer.destroy!
+        topic.reload
+
+        topic.inherit_auto_close_from_category(timer_type: :close)
+
+        return
+      end
+
+      if topic.open?
```

Comment:
can we move that code to line :12 to make it more concise?
Similar to combined logic from close_topic.rb
```
if topic.blank? || topic.open?
```